python/tchannel/tcurl.py

```diff
@@ -13,10 +17,46 @@
 
 
 @tornado.gen.coroutine
-def tcurl(host, endpoint, headers, body):
-
+def multi_tcurl(hostports, headers, bodies, profile=False):
     client = TChannel()
 
+    futures = [
+        tcurl(client, hostport, header, body)
+        for hostport, header, body
+        in getattr(itertools, 'izip', zip)(hostports, headers, bodies)
+    ]
+
+    start = time.time()
+
+    profiler = cProfile.Profile()
```

Comment:
Seems like you shouldn't incur the cost of instantiating this is `profile` is `None`.